prove_validator.go

```diff
@@ -45,18 +55,27 @@ func (epp *EigenPodProofs) ProveValidatorContainer(oracleBlockHeader *phase0.Bea
 	for i, validatorIndex := range validatorIndices {
 		VerifyValidatorFieldsCallParams.ValidatorIndices[i] = validatorIndex
 		// prove the validator fields against the beacon state
-		VerifyValidatorFieldsCallParams.ValidatorFieldsProofs[i], err = epp.ProveValidatorAgainstBeaconState(oracleBeaconState, beaconStateTopLevelRoots, validatorIndex)
+		VerifyValidatorFieldsCallParams.ValidatorFieldsProofs[i], err = epp.ProveValidatorAgainstBeaconState(oracleBeaconStateSlot, oracleBeaconStateValidators, beaconStateTopLevelRoots, validatorIndex)
 		if err != nil {
 			return nil, err
 		}
 
-		VerifyValidatorFieldsCallParams.ValidatorFields[i] = ConvertValidatorToValidatorFields(oracleBeaconState.Validators[validatorIndex])
+		VerifyValidatorFieldsCallParams.ValidatorFields[i] = ConvertValidatorToValidatorFields(oracleBeaconStateValidators[validatorIndex])
 	}
 
 	return VerifyValidatorFieldsCallParams, nil
 }
 
-func (epp *EigenPodProofs) ProveValidatorFields(oracleBlockHeader *phase0.BeaconBlockHeader, oracleBeaconState *deneb.BeaconState, validatorIndex uint64) (*StateRootProof, common.Proof, error) {
+func (epp *EigenPodProofs) ProveValidatorFields(oracleBlockHeader *phase0.BeaconBlockHeader, oracleBeaconState *spec.VersionedBeaconState, validatorIndex uint64) (*StateRootProof, common.Proof, error) {
```

Comment:
where is this used? epvbu? webserver?

Reply:
Its used in the CLI. I can get rid of it and switch those to use ProveValidatorContainers.

Reply:
ok I removed it, moved it to the generation package where it should be